File: february-ruby/lc/231_power_of_two.py
class Solution:
  def isPowerOfTwo(self, n: int) -> bool:
    return n != 0 and (n & (n - 1)) == 0


print(Solution().isPowerOfTwo(3))
print(Solution().isPowerOfTwo(0))
print(Solution().isPowerOfTwo(16))
print(Solution().isPowerOfTwo(4))
print(Solution().isPowerOfTwo(-3))


# class Solution {
#     public boolean isPowerOfTwo(int n) {
#         return n != 0 && (n & (n - 1)) == 0;
#     }
# }

File: february-ruby/lc/231_power_of_two.rb
# frozen_string_literal: true

def is_power_of_two(n)
  !n.zero? && (n & (n - 1)).zero?
end

p is_power_of_two(1)
p is_power_of_two(0)
p is_power_of_two(16)
p is_power_of_two(3)
p is_power_of_two(-2)